File: cpp/bc1828.cpp
#include<iostream>
#include<cstring>

using namespace std;

int main()
{
	int count;
	string fVal, sVal;
	
	cin >> count;
	
	for(int i = 1; i <= count; i++)
	{
		int winner = 0;
		cin >> fVal >> sVal;
			
		if(fVal == sVal) winner = 0;
		else if(fVal == "tesoura")
		{
			if(sVal == "papel" || sVal == "lagarto") winner = 1;
			else winner = 2;
		}
		else if(fVal == "papel")
		{
			if(sVal == "pedra" || sVal == "Spock") winner = 1;
			else winner = 2;
		}
		else if(fVal =="pedra")
		{
			if(sVal == "tesoura" || sVal == "lagarto") winner = 1;
			else winner = 2;
		}
		else if(fVal =="lagarto")
		{
			if(sVal == "Spock" || sVal == "papel") winner = 1;
			else winner = 2;
		}
		else
		{
			if(sVal == "pedra" || sVal == "tesoura") winner = 1;
			else winner = 2;
		}
	
		string res = "";
		
		if(winner == 1) res = "Bazinga!";
		else if(winner == 2) res = "Raj trapaceou!";
		else res = "De novo!";
		
		cout << "Caso #" << i << ": " << res << endl;
	}
}

File: python/bc1828.py
count = int(input())

for c in range(1, count+1):
    cVal = input().split(" ")

    fVal = cVal[0]
    sVal = cVal[1]

    winner = 0

    if fVal == sVal: winner = 0
    elif fVal == "tesoura":
        if sVal == "papel" or sVal == "lagarto": winner = 1
        else: winner = 2
    elif fVal == "papel":
        winner = 1 if sVal == "pedra" or sVal == "Spock" else 2
    elif fVal == "pedra":
        winner = 1 if sVal == "tesoura" or sVal == "lagarto" else 2
    elif fVal == "lagarto":
        winner = 1 if sVal == "Spock" or sVal == "papel" else 2
    else:
        winner = 1 if sVal == "pedra" or sVal == "tesoura" else 2

    res = ""

    if winner == 1: res = "Bazinga!"
    elif winner == 2: res = "Raj trapaceou!"
    else: res = "De novo!"

    print(f"Caso #{c}: {res}")
